### DISCUSSION TITLE
How can I use parameter bind_ip, server_i paramater?

### DISCUSSION THREAD

**Discussion by DaeyongKwon98** [ANLAB-KAIST/KENSv3#100] [Project Q&A 1. Application Layer (Socket)]
When I see socket programming in C(https://www.geeksforgeeks.org/socket-programming-cc/), there's no parameter like bind_ip, server_ip.

Homework instruction says we use these parameters in listen() and connect() call but I don't know how to use these parameters.

Also, when I tried to test my code with ./echo, I can only get pass/fail information.
With ./echo-non-kens, my result is blank like below.

Submitted answers:

I wonder why blank result appears.
Is there other ways to check(or test) my code, like using printf?

Thank you.

**Comment by whkim0** [comment]
1. It will be helpful to read thoroughly about how the server and client set parameters for `bind()`, `listen()`, `connect()`, especially `struct sockaddr`.
2. You can use `printf()` for debugging.
